### ISSUE TITLE
0.20.1 introduces false positive for trailing_whitespace

### ISSUE THREAD

**Issue by bsarrazin** [realm/SwiftLint#1732]
```
warning: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
```

Rules:
```
+-----------------------------------------+--------+-------------+------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------+
| identifier                              | opt-in | correctable | enabled in your config | configuration                                                                                                                                   |
+-----------------------------------------+--------+-------------+------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------+
| attributes                              | yes    | no          | yes                    | warning, always_on_same_line: ["@IBAction", "@NSManaged"], always_on_line_above: []                                                             |
| class_delegate_protocol                 | no     | no          | yes                    | warning                                                                                                                                         |
| closing_brace                           | no     | yes         | yes                    | warning                                                                                                                                         |
| closure_end_indentation                 | yes    | no          | yes                    | warning                                                                                                                                         |
| closure_parameter_position              | no     | no          | yes                    | warning                                                                                                                                         |
| closure_spacing                         | yes    | yes         | yes                    | warning                                                                                                                                         |
| colon                                   | no     | yes         | yes                    | warning, flexible_right_spacing: false, apply_to_dictionaries: true                                                                             |
| comma                                   | no     | yes         | yes                    | warning                                                                                                                                         |
| compiler_protocol_init                  | no     | no          | yes                    | warning                                                                                                                                         |
| conditional_returns_on_newline          | yes    | no          | no                     | warning                                                                                                                                         |
| control_statement                       | no     | no          | yes                    | warning                                                                                                                                         |
| custom_rules                            | no     | no          | yes                    | user-defined                                                                                                                                    |
| cyclomatic_complexity                   | no     | no          | yes                    | warning: 20, error: 30, ignores_case_statements: true                                                                                           |
| discarded_notification_center_observer  | no     | no          | yes                    | warning                                                                                                                                         |
| dynamic_inline                          | no     | no          | yes                    | error                                                                                                                                           |
| empty_count                             | yes    | no          | yes                    | error                                                                                                                                           |
| empty_enum_arguments                    | no     | yes         | yes                    | warning                                                                                                                                         |
| empty_parameters                        | no     | yes         | yes                    | warning                                                                                                                                         |
| empty_parentheses_with_trailing_closure | no     | yes         | yes                    | warning                                                                                                                                         |
| explicit_init                           | yes    | yes         | yes                    | warning                                                                                                                                         |
| explicit_top_level_acl                  | yes    | no          | no                     | warning                                                                                                                                         |
| explicit_type_interface                 | yes    | no          | no                     | warning                                                                                                                                         |
| extension_access_modifier               | yes    | no          | no                     | warning                                                                                                                                         |
| fatal_error_message                     | yes    | no          | yes                    | warning                                                                                                                                         |
| file_header                             | yes    | no          | yes                    | warning, required_string: None, required_pattern: None, forbidden_string: None, forbidden_pattern: None                                         |
| file_length                             | no     | no          | no                     | warning: 400, error: 1000                                                                                                                       |
| first_where                             | yes    | no          | yes                    | warning                                                                                                                                         |
| for_where                               | no     | no          | yes                    | warning                                                                                                                                         |
| force_cast                              | no     | no          | no                     | error                                                                                                                                           |
| force_try                               | no     | no          | yes                    | error                                                                                                                                           |
| force_unwrapping                        | yes    | no          | no                     | warning                                                                                                                                         |
| function_body_length                    | no     | no          | yes                    | warning: 150, error: 200                                                                                                                        |
| function_parameter_count                | no     | no          | yes                    | warning: 5, error: 8                                                                                                                            |
| generic_type_name                       | no     | no          | yes                    | (min_length) w/e: 1/0, (max_length) w/e: 20/1000, excluded: [], allowed_symbols: [], validates_start_with_lowercase: true                       |
| identifier_name                         | no     | no          | yes                    | (min_length) w: 3, (max_length) w/e: 50/75, excluded: ["ID", "id", "ok", "op", "up"], allowed_symbols: [], validates_start_with_lowercase: true |
| implicit_getter                         | no     | no          | yes                    | warning                                                                                                                                         |
| implicit_return                         | yes    | yes         | yes                    | warning                                                                                                                                         |
| implicitly_unwrapped_optional           | yes    | no          | no                     | warning, mode: allExceptIBOutlets                                                                                                               |
| large_tuple                             | no     | no          | yes                    | warning: 2, error: 3                                                                                                                            |
| leading_whitespace                      | no     | yes         | yes                    | warning                                                                                                                                         |
| legacy_cggeometry_functions             | no     | yes         | yes                    | warning                                                                                                                                         |
| legacy_constant                         | no     | yes         | yes                    | warning                                                                                                                                         |
| legacy_constructor                      | no     | yes         | yes                    | warning                                                                                                                                         |
| legacy_nsgeometry_functions             | no     | yes         | yes                    | warning                                                                                                                                         |
| line_length                             | no     | no          | yes                    | warning: 500, error: 1000, ignores urls: true, ignores function declarations: true, ignores comments: true                                      |
| mark                                    | no     | yes         | yes                    | warning                                                                                                                                         |
| multiline_parameters                    | yes    | no          | no                     | warning                                                                                                                                         |
| nesting                                 | no     | no          | yes                    | (type_level) w: 3, (statement_level) w: 5                                                                                                       |
| nimble_operator                         | yes    | yes         | no                     | warning                                                                                                                                         |
| no_extension_access_modifier            | yes    | no          | no                     | error                                                                                                                                           |
| notification_center_detachment          | no     | no          | no                     | warning                                                                                                                                         |
| number_separator                        | yes    | yes         | yes                    | warning, minimum_length: 5                                                                                                                      |
| object_literal                          | yes    | no          | no                     | warning, image_literal: true, color_literal: true                                                                                               |
| opening_brace                           | no     | yes         | yes                    | warning                                                                                                                                         |
| operator_usage_whitespace               | yes    | yes         | yes                    | warning                                                                                                                                         |
| operator_whitespace                     | no     | no          | yes                    | warning                                                                                                                                         |
| overridden_super_call                   | yes    | no          | yes                    | warning, excluded: [[]], included: [["*"]]                                                                                                      |
| private_outlet                          | yes    | no          | no                     | warning, allow_private_set: false                                                                                                               |
| private_unit_test                       | no     | no          | yes                    | warning: XCTestCase                                                                                                                             |
| prohibited_super_call                   | yes    | no          | yes                    | warning, excluded: [[]], included: [["*"]]                                                                                                      |
| protocol_property_accessors_order       | no     | yes         | yes                    | warning                                                                                                                                         |
| redundant_discardable_let               | no     | yes         | yes                    | warning                                                                                                                                         |
| redundant_nil_coalescing                | yes    | yes         | yes                    | warning                                                                                                                                         |
| redundant_optional_initialization       | no     | yes         | yes                    | warning                                                                                                                                         |
| redundant_string_enum_value             | no     | no          | yes                    | warning                                                                                                                                         |
| redundant_void_return                   | no     | yes         | yes                    | warning                                                                                                                                         |
| return_arrow_whitespace                 | no     | yes         | yes                    | warning                                                                                                                                         |
| shorthand_operator                      | no     | no          | yes                    | warning                                                                                                                                         |
| sorted_imports                          | yes    | no          | yes                    | warning                                                                                                                                         |
| statement_position                      | no     | yes         | yes                    | (statement_mode) default, (severity) warning                                                                                                    |
| switch_case_on_newline                  | yes    | no          | yes                    | warning                                                                                                                                         |
| syntactic_sugar                         | no     | no          | yes                    | warning                                                                                                                                         |
| todo                                    | no     | no          | no                     | warning                                                                                                                                         |
| trailing_comma                          | no     | yes         | yes                    | warning, mandatory_comma: false                                                                                                                 |
| trailing_newline                        | no     | yes         | yes                    | warning                                                                                                                                         |
| trailing_semicolon                      | no     | yes         | yes                    | warning                                                                                                                                         |
| trailing_whitespace                     | no     | yes         | yes                    | warning, ignores_empty_lines: true, ignores_comments: false                                                                                     |
| type_body_length                        | no     | no          | yes                    | warning: 500, error: 5000                                                                                                                       |
| type_name                               | no     | no          | yes                    | (min_length) w: 3, (max_length) w/e: 70/75, excluded: ["ID", "OK", "Ok", "id"], allowed_symbols: [], validates_start_with_lowercase: true       |
| unused_closure_parameter                | no     | yes         | no                     | warning                                                                                                                                         |
| unused_enumerated                       | no     | no          | no                     | warning                                                                                                                                         |
| unused_optional_binding                 | no     | no          | no                     | warning, ignore_optional_try: false                                                                                                             |
| valid_ibinspectable                     | no     | no          | yes                    | warning                                                                                                                                         |
| vertical_parameter_alignment            | no     | no          | yes                    | warning                                                                                                                                         |
| vertical_parameter_alignment_on_call    | yes    | no          | yes                    | warning                                                                                                                                         |
| vertical_whitespace                     | no     | yes         | yes                    | warning, max_empty_lines: 1                                                                                                                     |
| void_return                             | no     | yes         | yes                    | warning                                                                                                                                         |
| weak_delegate                           | no     | no          | yes                    | warning                                                                                                                                         |
+-----------------------------------------+--------+-------------+------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------+

```

**Comment by marcelofabri** [comment]
@bsarrazin Could you share an example that shows the false positive?

**Comment by bsarrazin** [comment]
![2017-08-01 at 10 13 am](https://user-images.githubusercontent.com/1077441/28829641-6a565218-76a2-11e7-925a-103bb729b907.png)

Here you go.
Same problem with `force_cast`, same problem with `line_length`.

Returning to 0.20.0 fixes the problem.

**Comment by marcelofabri** [comment]
I highly doubt it's an issue with any of this rules. It's more likely a consequence of [this fix](https://github.com/realm/SwiftLint/issues/1631).

Do you have nested configuration files? How are you invoking SwiftLint?

**Comment by bsarrazin** [comment]
@marcelofabri we invoke swiftlint through a build phase:
```
${PODS_ROOT}/SwiftLint/swiftlint lint --path $TARGET_NAME
```

**Comment by bsarrazin** [comment]
We have no nested config files.

**Comment by marcelofabri** [comment]
Is your file structure like this?

```
- ProjectFolder
    - .swiftlint.yml
    - SomeTargetFolder
    - AnotherTargetFolder
```

If so, I'd say that actually 0.20.0 introduced a bug that made it work, but it was fixed on 0.20.1. Older versions should have the same behavior.

The solution would be to use `--config` as well:

```
${PODS_ROOT}/SwiftLint/swiftlint lint --path $TARGET_NAME --config .swiftlint.yml
```

**Comment by bsarrazin** [comment]
well this is interesting...
Adding the config flag now causes this:
```
$ Pods/SwiftLint/swiftlint lint --config .swiftlint.yml --path MyKit
fatal error: Could not read configuration file at path '/Users/ben/Development/Projects/[redacted]/[redacted]/MyKit/.swiftlint.yml': file /Users/marcelofabri/Westfield/SwiftLint/Source/SwiftLintFramework/Models/Configuration.swift, line 167
/Users/ben/Development/Projects/[redacted]/[redacted]/MyKit/.swiftlint.yml:File not found.
[1]    81491 illegal hardware instruction  Pods/SwiftLint/swiftlint lint --config .swiftlint.yml --path MyKit
```

**Comment by bsarrazin** [comment]
I can fix the path being passed in the `--config` flag, but seeing the local path on your computer might not be the best thing 👅

**Comment by marcelofabri** [comment]
Oh right, this is another issue that was fixed on master 😅 

Sorry about that. Using the absolute path is a workaround for now (or using master)

**Comment by bsarrazin** [comment]
when is the next release schedule for?

**Comment by marcelofabri** [comment]
No promises, but I think it should be soon. Could you try using an absolute path just to be sure it works?

**Comment by bsarrazin** [comment]
Using an absolute path works.
thanks for the help Marcelo!

**Comment by marcelofabri** [comment]
Thanks for your patience and sorry about these issues 😬

**Comment by ParulThoughtworks** [comment]
I am facing following problem with **swiftlint version - 0.20.1** 

> Invalid configuration for 'line_length'. Falling back to default.

> Would like to know if the rule is renamed? If so i can't find it in swiftlint rules.

**Attaching the screenshots below**

> Swiftlint warning

<img width="323" alt="screen shot 2017-09-01 at 1 17 37 pm" src="https://user-images.githubusercontent.com/15243286/29960285-aeab1b72-8f18-11e7-9d68-552203697cc3.png">


> Swiftlint.yml

<img width="192" alt="screen shot 2017-09-01 at 1 24 28 pm" src="https://user-images.githubusercontent.com/15243286/29960375-00bad718-8f19-11e7-8fd0-9ca5d34cf5f9.png">

> Swiftlint rules

<img width="1413" alt="screen shot 2017-09-01 at 1 25 00 pm" src="https://user-images.githubusercontent.com/15243286/29960404-190f5906-8f19-11e7-92f3-f088e4fc4c78.png">

**Comment by marcelofabri** [comment]
@ParulKSharda please update to the latest version. If the issue persists, open a new issue.
